### PR TITLE
Add deletion_protection field for cloudExadataInfrastructure resource

### PR DESCRIPTION
**Pull request by tulika-aakriti**
part of : https://github.com/hashicorp/terraform-provider-google/issues/20284

**Release Note Template for Downstream PRs (will be copied)**

See [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) for guidance.

```release-note: enhancement
oracledatabase: added `deletion_protection` field to `google_oracle_database_cloud_exadata_infrastructure`
```

### CODE REVIEW

File: mmv1/templates/terraform/pre_delete/oracledatabase_cloud_exadata_infrastructure.go.tmpl [hunk 1]
```diff
@@ -0,0 +1,3 @@
+if d.Get("deletion_protection").(bool) {
+	return fmt.Errorf("cannot destroy instance without setting deletion_protection=false and running `terraform apply`")
```
Comment (rileykarson) [inline mmv1/templates/terraform/pre_delete/oracledatabase_cloud_exadata_infrastructure.go.tmpl:2]:
```suggestion
	return fmt.Errorf("cannot destroy google_oracle_database_cloud_exadata_infrastructure resource %q without setting deletion_protection=false and running `terraform apply`", d.Id())
```

Let's make the error message more specific to the resource, and call out which it is.

(may not run as written, but should be easy to fix if it doesn't)

**Review by rileykarson — COMMENTED** [review]
Can you amend those two tests as well? Thanks!

**Comment by tulika-aakriti** [comment]
> Can you amend those two tests as well? Thanks!

I update the instance in the tests, rest field for other instance is added in https://github.com/GoogleCloudPlatform/magic-modules/pull/12297

**Comment by tulika-aakriti** [comment]
@rileykarson Please approve the test build.

**Review by slevenick — APPROVED** [review]

**Pull request merged**